### PR TITLE
Add a unit test from aspnetcore

### PR DESCRIPTION
**Pull request by pgovind**
Add a unit test that covers a critical scenario from ASP.NET. This scenario was identified when #51508 was merged in and caused a regression in ASP.NET, as reported when the change was backported to [preview 7](https://github.com/dotnet/runtime/pull/55960#issuecomment-883589136). That regression was first observed in dotnet/aspnetcore#34491.

Rule: `(.*)/(.*).aspx`
Input: `/pages/homepage.aspx`

**Expected** backreferences:
1. /pages/homepage.aspx
2. /pages
3. homepage

**Actual** backreferences: None, didn't match

This scenario represents multiple `.*` groups within a pattern and being able to capture all expected backreferences.

### CODE REVIEW

File: src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs [hunk 1]
```diff
@@ -823,6 +823,9 @@ public static IEnumerable<object[]> Groups_Basic_TestData()
             // Empty Match
             yield return new object[] { null, @"([a*]*)+?$", "ab", RegexOptions.None, new string[] { "", "" } };
             yield return new object[] { null, @"(a*)+?$", "b", RegexOptions.None, new string[] { "", "" } };
+
+            // A unit test from aspnetcore
```
Comment (stephentoub) [inline src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs:827]: Can we instead say what's special about this particular pattern? i.e. this obviously covers something that literally none of our other tests do... what is it?

File: src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs [hunk 2]
```diff
@@ -582,6 +582,53 @@ public static IEnumerable<object[]> Groups_Basic_TestData()
             yield return new object[] { null, @"([a-z]*)([\w])", "cat", RegexOptions.IgnoreCase, new string[] { "cat", "ca", "t" } };
             yield return new object[] { null, @"^([a-z]*)([\w])$", "cat", RegexOptions.IgnoreCase, new string[] { "cat", "ca", "t" } };
 
+            foreach (RegexOptions options in new[] { RegexOptions.None, RegexOptions.Compiled })
```
Comment (campersau) [inline src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs:585]: `options` is unused.

Comment (stephentoub) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs:585]: This loop is unnecessary entirely, actually.  This member data is only used with one test, that already invokes the core routine twice for each input, once without or'ing in Compiled and one with:
https://github.com/dotnet/runtime/blob/ffa17d6c569ec75ca7f583506a699428b850a3f5/src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs#L932-L933

Comment (jeffhandley) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs:585]: Oops; I meant to use it in each of the cases. Thanks!

Comment (jeffhandley) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs]: None and Compiled were already covered by separate calls in the test. The loop was unneeded; removed.

File: src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs [hunk 3]
```diff
@@ -582,6 +582,50 @@ public static IEnumerable<object[]> Groups_Basic_TestData()
             yield return new object[] { null, @"([a-z]*)([\w])", "cat", RegexOptions.IgnoreCase, new string[] { "cat", "ca", "t" } };
             yield return new object[] { null, @"^([a-z]*)([\w])$", "cat", RegexOptions.IgnoreCase, new string[] { "cat", "ca", "t" } };
 
+            // Backtracking with multiple (.*) groups -- important ASP.NET scnario
```
Comment (jeffhandley) [inline src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs:585]:
```suggestion
            // Backtracking with multiple (.*) groups -- important ASP.NET scenario
```

Comment (jeffhandley) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs]: These commits were sloppy; sorry for the noise.

**Comment by pgovind** [comment]
GH won't let me approve this PR since I opened it. The changes look good to me. There are interesting patterns to add with `.?` too, but I'll do that on my next pass through these files.

**Review by jeffhandley — APPROVED** [review]

**Comment by jeffhandley** [comment]
Thanks, @pgovind. Good thinking on `.?`, but per your comment, I won't add them here/now.

I noticed that there's very little coverage of _negative scenarios_ where matches are unsuccessful. One of the mistakes I made yesterday was assuming I could cover such scenarios within the group tests, but that's not in place--the test method assumes every scenario is a successful match. There are a few other tests in this feature area for unsuccessful matches, but nowhere near the coverage we have in these group tests. I held back from updating this particular test suite to capture such scenarios.

**Comment by jeffhandley** [comment]
Failure is unrelated and infrastructural.

**Pull request merged**